src/ui/hooks/useContactHook.ts

```diff
@@ -0,0 +1,271 @@
+import { isEmpty } from 'lodash';
+import { useCallback } from 'react';
+
+import { type Contact } from '@/shared/types/network-types';
+import { type WalletAddress } from '@/shared/types/wallet-types';
+import { withPrefix, isValidEthereumAddress } from '@/shared/utils/address';
+import { useProfileHook } from '@/ui/hooks/useProfileHook';
+import { useContactStore } from '@/ui/stores/contactStore';
+import { useWallet } from '@/ui/utils';
+
+const DEFAULT_CONTACT: Contact = {
+  address: '',
+  id: 0,
+  contact_name: '',
+  avatar: '',
+  domain: {
+    domain_type: 999,
+    value: '',
+  },
+};
+
+export function useContactHook() {
```

Comment:
No need to add Hook to the end of the name. This would should be `useContacts` plural. The use at the beginning should be good enough to let us know it is a hook
```suggestion
export function useContacts() {
```